tests/test_main.py

```diff
@@ -1359,23 +1366,25 @@ def foo():
 lamb = lambda: 0
 
 
-assert tester([1, 2, 3]) == [1, 2, 3]
+def test_tester():
```

Comment:
The reason this stuff was global is that there was a time when global vs local mattered, and so this was there to test that things still worked globally. It's probably fine to move past that now, and the global stuff *was* irritating, but as we've seen with the recent decorator bug, weird things happen sometimes with bytecode.

Reply:
interesting, my problem was that i got a problem during test collection
```
================================================= ERRORS =================================================
__________________________________ ERROR collecting tests/test_main.py ___________________________________
tests/test_main.py:1370: in <module>
    assert tester([1, 2, 3]) == [1, 2, 3]
tests/utils.py:51: in __call__
    ex = self.get_executing(inspect.currentframe().f_back)
tests/utils.py:39: in get_executing
    return Source.executing(frame)
executing/executing.py:337: in executing
    node_finder = NodeFinder(frame, stmts, tree, lasti, source)
executing/executing.py:633: in __init__
    matching = list(self.matching_nodes(exprs))
executing/executing.py:699: in matching_nodes
    original_index = only(
executing/executing.py:167: in only
    raise NotOneValueFound('Expected one value, found 0')
E   executing.executing.NotOneValueFound: Expected one value, found 0
======================================== short test summary info =========================================
ERROR tests/test_main.py - executing.executing.NotOneValueFound: Expected one value, found 0
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Does it look familiar to you?
I look for the reason and try to make it work.

Reply:
Sounds like either:

1. pytest is still rewriting assertions
2. assertions have problems even without pytest rewriting
3. something at the global level really does make a difference still

Reply:
I have `test_main.py` minimized to the following

```
# -*- coding: utf-8 -*-
"""

assert rewriting will break executing
PYTEST_DONT_REWRITE
"""
import pytest


from tests.utils import tester


import inspect
frame=inspect.currentframe()
import dis
dis.dis(frame.f_code)

assert tester([1, 2, 3]) == [1, 2, 3]

```

Which fails with the exception from above.
The assert rewriting is correct disabled.
The thing is that removing `import pytest` makes it work.
`pytest` is not used anywhere and the generated bytecode looks identical except the missing import.

I will stop here. Maybe I have tomorrow an idea ... or you.

Reply:
Ah, that makes sense.

https://github.com/alexmojaki/executing/blob/11648389b9c8c3b59ff1137f64d1fc7c0651cb40/executing/executing.py#L543-L550

Seems like we would need a better way to detect (for the sake of avoiding/ignoring) pytest assertion rewriting.

Reply:
I check the bytecode for LOAD ops with "@py_assert" as variable name. What do you think?